### PR TITLE
Feature/oe 5540

### PR DESCRIPTION
**Pull request by himanshupatel79**
This include changes of case OE-5539 and OE-5540.

### CODE REVIEW

File: protected/migrations/m150505_113604_doctor_grade.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+		array('id'=> 6, 'grade' => 'Senior House Officer'),
+		array('id'=> 7, 'grade' => 'House officer'),
+	);
+
+	public function up()
+	{
+		$this->createTable(
+			'doctor_grade',
+			array(
+				'id' => 'int(3) not null',
```
Comment (vetusko) [inline protected/migrations/m150505_113604_doctor_grade.php]: I think we should use auto_increment and primary_key for the id fields in general

File: protected/migrations/m150505_113604_doctor_grade.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	private $grades = array(
+		array('id'=> 1, 'grade' => 'Consultant'),
+		array('id'=> 2, 'grade' => 'Associate specialist'),
+		array('id'=> 3, 'grade' => 'Trust doctor'),
+		array('id'=> 4, 'grade' => 'Fellow'),
+		array('id'=> 5, 'grade' => 'Specialist Registrar'),
+		array('id'=> 6, 'grade' => 'Senior House Officer'),
+		array('id'=> 7, 'grade' => 'House officer'),
+	);
+
```
Comment (vetusko) [inline protected/migrations/m150505_113604_doctor_grade.php]: we should use auto_increment for the ID, so id fields should be empty for the inserts

File: protected/migrations/m150505_113604_doctor_grade.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	public function up()
+	{
+		$this->createTable(
+			'doctor_grade',
+			array(
+				'id' => 'int(3) not null',
+				'grade' => 'varchar(100) not null',
```
Comment (vetusko) [inline protected/migrations/m150505_113604_doctor_grade.php]: in the model's rules it's max 250, need to change this to varchar(250), or the rule to max 100

File: protected/models/DoctorGrade.php [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+	/**
+	 * @return array validation rules for model attributes.
+	 */
+	public function rules()
+	{
+		// NOTE: you should only define rules for those attributes that
+		// will receive user inputs.
+		return array(
+			array('grade', 'required'),
+			array('grade', 'length', 'max'=>250),
```
Comment (vetusko) [inline protected/models/DoctorGrade.php]: change this to max=100 or change the table definition

File: protected/models/DoctorGrade.php [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+			// The following rule is used by search().
+			// Please remove those attributes that should not be searched.
+			array('id, grade, registration_code', 'safe'),
```
Comment (vetusko) [inline protected/models/DoctorGrade.php]: there is no registration_code field in the doctor_grade table

File: protected/models/DoctorGrade.php [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+	/**
+	 * @return array relational rules.
+	 */
+	public function relations()
+	{
+		// NOTE: you may need to adjust the relation name and the related
+		// class name for the relations automatically generated below.
+		return array(
+			'user_grade' => array(self::BELONGS_TO, 'UserGrade', 'grade'),
```
Comment (vetusko) [inline protected/models/DoctorGrade.php]: The model here should be User, not UserGrade, there is no UserGrade module exists

File: protected/models/DoctorGrade.php [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+	 */
+	public function search()
+	{
+		// Warning: Please modify the following code to remove attributes that
+		// should not be searched.
+
+		$criteria=new CDbCriteria;
+
+		$criteria->compare('id',$this->id,true);
+		$criteria->compare('grade',$this->medication_id,true);
```
Comment (vetusko) [inline protected/models/DoctorGrade.php]: medication_id? why is it here?

File: protected/migrations/m150505_113733_add_grade_reg_code.php [hunk 8]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+class m150505_113733_add_grade_reg_code extends OEMigration
+{
+	public function up()
+	{
+
+		$this->addColumn('user', 'grade', 'int(3) null default null');
+		$this->addColumn('user_version', 'grade', 'int(3) null default null');
+
```
Comment (vetusko) [inline protected/migrations/m150505_113733_add_grade_reg_code.php]: that should be grade_id in both

File: protected/migrations/m150505_113733_add_grade_reg_code.php [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+		$this->addColumn('user_version', 'registration_code', 'varchar(250) null default null');
+	}
+
+	public function down()
+	{
+		$this->dropForeignKey('user_doctor_grade_id_fk','user');
+
+		$this->dropColumn('user', 'grade');
+		$this->dropColumn('user_version', 'grade');
+
```
Comment (vetusko) [inline protected/migrations/m150505_113733_add_grade_reg_code.php]: should be grade_id

File: protected/models/User.php [hunk 10]
```diff
@@ -123,6 +123,7 @@ public function relations()
 			'preferred_firms' => array(self::HAS_MANY, 'Firm', 'firm_id', 'through' => 'firm_preferences', 'order' => 'firm_preferences.position DESC', 'limit' => 6),
 			'firmSelections' => array(self::MANY_MANY, 'Firm', 'user_firm(firm_id, user_id)', 'condition' => 'firmSelections.active = 1', 'order' => 'name asc'),
 			'siteSelections' => array(self::MANY_MANY, 'Site', 'user_site(site_id, user_id)', 'order' => 'name asc'),
+			'grade' => array(self::BELONGS_TO, 'doctor_grade', 'id'),
```
Comment (vetusko) [inline protected/models/User.php]: Should be:  'grade' => array(self::BELONGS_TO, 'DoctorGrade', 'grade_id'),

File: protected/views/admin/adduser.php [hunk 11]
```diff
@@ -45,7 +45,9 @@
 		<?php echo $form->radioBoolean($user,'active')?>
 		<?php echo $form->radioBoolean($user,'global_firm_rights')?>
 		<?php echo $form->radioBoolean($user,'is_doctor')?>
-		<?php echo $form->passwordField($user,'password',array('autocomplete'=>Yii::app()->params['html_autocomplete']))?>
+		<?php echo $form->textField($user,'registration_code',array('autocomplete'=>Yii::app()->params['html_autocomplete']))?>
+	    <?php echo $form->dropDownList($user,'grade', CHtml::listData(DoctorGrade::model()->findAll(array('order' => 'id')), 'id', 'grade'));?>
```
Comment (vetusko) [inline protected/views/admin/adduser.php]: should be grade_id here as well

File: protected/views/admin/edituser.php [hunk 12]
```diff
@@ -41,7 +41,9 @@
 		<?php echo $form->radioBoolean($user,'active')?>
 		<?php echo $form->radioBoolean($user,'global_firm_rights')?>
 		<?php echo $form->radioBoolean($user,'is_doctor')?>
-		<?php echo $form->passwordField($user,'password',array('autocomplete'=>Yii::app()->params['html_autocomplete']))?>
+	    <?php echo $form->textField($user,'registration_code',array('autocomplete'=>Yii::app()->params['html_autocomplete']))?>
+	    <?php echo $form->dropDownList($user,'grade', CHtml::listData(DoctorGrade::model()->findAll(array('order' => 'id')), 'id', 'grade'));?>
```
Comment (vetusko) [inline protected/views/admin/edituser.php]: grade_id

File: protected/migrations/m150505_113604_doctor_grade.php [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+			)
```
Comment (vetusko) [inline protected/migrations/m150505_113604_doctor_grade.php]: display_order field required

File: protected/migrations/m150505_113604_doctor_grade.php (hunk 2 above)
Comment (vetusko) [inline protected/migrations/m150505_113604_doctor_grade.php]: need to change id to display_order

**Comment by himanshupatel79** [comment]
Added Code Review Corrections.

File: protected/models/User.php [hunk 14]
```diff
@@ -123,6 +123,7 @@ public function relations()
 			'preferred_firms' => array(self::HAS_MANY, 'Firm', 'firm_id', 'through' => 'firm_preferences', 'order' => 'firm_preferences.position DESC', 'limit' => 6),
 			'firmSelections' => array(self::MANY_MANY, 'Firm', 'user_firm(firm_id, user_id)', 'condition' => 'firmSelections.active = 1', 'order' => 'name asc'),
 			'siteSelections' => array(self::MANY_MANY, 'Site', 'user_site(site_id, user_id)', 'order' => 'name asc'),
+			'grade' => array(self::BELONGS_TO, 'doctor_grade', 'doctor_grade_id'),
```
Comment (vetusko) [inline protected/models/User.php]: you should use DoctorGrade here instead of doctor_grade

**Pull request merged**